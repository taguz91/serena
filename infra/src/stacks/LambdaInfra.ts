import { CfnOutput, Stack } from "aws-cdk-lib";
import { Construct } from "constructs";

import { CheckStudentRole } from "./roles/CheckStudentRole";
import { FindStudentRole } from "./roles/FindStudentRole";
import { Environment } from "src/interfaces";

interface Props {
  env: Environment;
}

export default class LambdaInfra extends Stack {
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id, props);

    const findStudentRole = new FindStudentRole(this, "FindStudentRole", {
      s3Arn: props.env.s3Arn,
    });

    const checkStudent = new CheckStudentRole(this, "CheckStudentRole", {
      s3Arn: props.env.s3Arn,
    });

    // outputs
    new CfnOutput(this, "FindStudentRoleOutput", {
      value: findStudentRole.arn,
      description: "Role for find student role",
      exportName: this.exportNameWrapper("FindStudentRoleArn"),
    });

    new CfnOutput(this, "CheckStudentRoleOutput", {
      value: checkStudent.arn,
      description: "Role for check student role",
      exportName: this.exportNameWrapper("CheckStudentRoleArn"),
    });
  }

  private exportNameWrapper(name: string) {
    return `${this.stackName}-${name}`;
  }
}
