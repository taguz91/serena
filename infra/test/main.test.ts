import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import LambdaInfra from "../src/stacks/LambdaInfra";

test("Snapshot", () => {
  const app = new App();
  const stack = new LambdaInfra(app, "test", {
    env: {
      s3Arn: "arn:aws:s3:::test",
      stage: "dev",
      region: "us-east-1",
      account: "test",
    },
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
