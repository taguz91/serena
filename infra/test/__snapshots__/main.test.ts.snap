// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Outputs": {
    "CheckStudentRoleOutput": {
      "Description": "Role for check student role",
      "Export": {
        "Name": "test-CheckStudentRoleArn",
      },
      "Value": {
        "Fn::GetAtt": [
          "CheckStudentRole6C985858",
          "Arn",
        ],
      },
    },
    "FindStudentRoleOutput": {
      "Description": "Role for find student role",
      "Export": {
        "Name": "test-FindStudentRoleArn",
      },
      "Value": {
        "Fn::GetAtt": [
          "FindStudentRole32260A61",
          "Arn",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CheckStudentRole6C985858": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "Role for check the student emotion in the class",
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
          {
            "Ref": "CheckStudentRoleFindStudentPolicy2D3A0E96",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CheckStudentRoleFindStudentPolicy2D3A0E96": {
      "Properties": {
        "Description": "",
        "Path": "/",
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "S3:GetObject",
              "Effect": "Allow",
              "Resource": "arn:aws:s3:::test/*",
              "Sid": "AllowReadFilesFromS3",
            },
            {
              "Action": "rekognition:DetectFaces",
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "AllowDetectFacesEmotions",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "FindStudentRole32260A61": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "Role for find the student, in inscription records",
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
          {
            "Ref": "FindStudentRoleFindStudentPolicyDFC73662",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FindStudentRoleFindStudentPolicyDFC73662": {
      "Properties": {
        "Description": "",
        "Path": "/",
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "S3:GetObject",
              "Effect": "Allow",
              "Resource": "arn:aws:s3:::test/*",
              "Sid": "AllowReadFilesFromS3",
            },
            {
              "Action": "rekognition:CompareFaces",
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "AllowCompareFaces",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
